src/spatial/modules/main/spatial_functions_scalar.cpp

```diff
@@ -6256,6 +6257,239 @@ struct ST_Hilbert {
 	}
 };
 
+//======================================================================================================================
+// ST_InteriorRingN
+//======================================================================================================================
+
+struct ST_InteriorRingN {
+
+    //------------------------------------------------------------------------------------------------------------------
+    // GEOMETRY
+    //------------------------------------------------------------------------------------------------------------------
+    static void ExecuteGeometry(DataChunk &args, ExpressionState &state, Vector &result) {
+        auto &lstate = LocalState::ResetAndGet(state);
+
+        BinaryExecutor::Execute<string_t, int64_t, string_t>(
+            args.data[0], args.data[1], result, args.size(),
+            [&](const string_t &blob, const int64_t &n) {
+                sgl::geometry geom;
+                lstate.Deserialize(blob, geom);
+
+                // ---- validate geometry ----
+                if (geom.get_type() != sgl::geometry_type::POLYGON) {
+                    // same semantics as ST_ExteriorRing: return NULL for non-polygons
+                    return string_t{};
+                }
+
+                if (geom.is_empty()) {
+                    // empty polygon → empty linestring
+                    const sgl::geometry empty_ls(sgl::geometry_type::LINESTRING,
+                                                 geom.has_z(), geom.has_m());
+                    return lstate.Serialize(result, empty_ls);
+                }
+
+                if (n < 1) {
+                    // invalid index → NULL
+                    return string_t{};
+                }
+
+                const idx_t num_parts = geom.get_part_count(); // includes shell
+                // parts: 0 = exterior, 1..n = interior rings
+                const idx_t num_interior = num_parts > 0 ? num_parts - 1 : 0;
+
+                if ((idx_t)n > num_interior) {
```

Comment:
Minor nitpick, but we generally prefer c++-style `static_cast` or `UnsafeNumericCast<>` over c-style casts